packages/js/data/src/products/types.ts

```diff
@@ -37,32 +43,65 @@ export type Product< Status = ProductStatus, Type = ProductType > = Omit<
 	description: string;
 	short_description: string;
 	sku: string;
+	date_on_sale_from: string | null;
+	date_on_sale_from_gmt: string | null;
+	date_on_sale_to: string | null;
+	date_on_sale_to_gmt: string | null;
+	virtual: boolean;
+	downloadable: boolean;
+	downloads: ProductDownload[];
+	download_limit: number;
+	download_expiry: number;
+	external_url: string;
+	button_text: string;
+	tax_status: 'taxable' | 'shipping' | 'none';
+	tax_class: 'standard' | 'reduced-rate' | 'zero-rate' | undefined;
+	manage_stock: boolean;
+	stock_quantity: number;
+	stock_status: 'instock' | 'outofstock' | 'onbackorder';
+	backorders: 'no' | 'notify' | 'yes';
 	price: string;
+	price_html: string;
 	regular_price: string;
 	sale_price: string;
+	on_sale: boolean;
+	purchasable: boolean;
+	total_sales: number;
+	backorders_allowed: boolean;
+	backordered: boolean;
+	shipping_required: boolean;
+	shipping_taxable: boolean;
+	shipping_class_id: number;
+	average_rating: string;
+	rating_count: number;
+	related_ids: number[];
+	variations: number[];
 };
 
-export type ReadOnlyProperties =
-	| 'id'
-	| 'permalink'
-	| 'date_created'
-	| 'date_created_gmt'
-	| 'date_modified'
-	| 'date_modified_gmt'
-	| 'price'
-	| 'price_html'
-	| 'on_sale'
-	| 'purchasable'
-	| 'total_sales'
-	| 'backorders_allowed'
-	| 'backordered'
-	| 'shipping_required'
-	| 'shipping_taxable'
-	| 'shipping_class_id'
-	| 'average_rating'
-	| 'rating_count'
-	| 'related_ids'
-	| 'variations';
+export const productReadOnlyProperties = [
+	'id',
+	'permalink',
+	'date_created',
+	'date_created_gmt',
+	'date_modified',
+	'date_modified_gmt',
+	'price',
+	'price_html',
+	'on_sale',
+	'purchasable',
+	'total_sales',
+	'backorders_allowed',
+	'backordered',
+	'shipping_required',
+	'shipping_taxable',
+	'shipping_class_id',
+	'average_rating',
+	'rating_count',
+	'related_ids',
+	'variations',
+] as const;
```

Comment:
Moved these to a array, so I can use them more easily to remove these properties when updating or copying a product.